From bdb5b13aaa20946ac9ccebffe55e7747f4ad54ec Mon Sep 17 00:00:00 2001
From: Simon McVittie <simon.mcvittie@collabora.co.uk>
Date: Tue, 25 Oct 2016 20:01:10 +0100
Subject: [PATCH 6/6] testbed: mount / rw before invoking dpkg/apt

This assumes we can "apt-get update" without remounting (in other words,
/var is always rw), but might need to remount the root read/write
before proceeding.

Signed-off-by: Simon McVittie <simon.mcvittie@collabora.co.uk>
---
 lib/adt_binaries.py | 1 +
 lib/adt_run_args.py | 1 +
 lib/adt_testbed.py  | 1 +
 3 files changed, 3 insertions(+)

diff --git a/lib/adt_binaries.py b/lib/adt_binaries.py
index c723c37..21c9e13 100644
--- a/lib/adt_binaries.py
+++ b/lib/adt_binaries.py
@@ -125,6 +125,7 @@ class DebBinaries:
                     adtlog.debug('Binaries: publish reinstall needs ' + pkg)
 
         if pkgs_reinstall:
+            self.testbed.check_exec(['mount', '-o', 'remount,rw', '/'])
             rc = self.testbed.execute(
                 ['apt-get', '--quiet', '-o', 'Debug::pkgProblemResolver=true',
                  '-o', 'APT::Get::force-yes=true',
diff --git a/lib/adt_run_args.py b/lib/adt_run_args.py
index 7f6aa3e..2e50562 100644
--- a/lib/adt_run_args.py
+++ b/lib/adt_run_args.py
@@ -267,6 +267,7 @@ details.'''
                          action='append_const',
                          const='(apt-get update || (sleep 15; apt-get update)'
                          ' || (sleep 60; apt-get update) || false)'
+                         ' && mount -o remount,rw /'
                          ' && $(which eatmydata || true) apt-get dist-upgrade -y -o '
                          'Dpkg::Options::="--force-confnew"',
                          help='Run apt update/dist-upgrade before the tests')
diff --git a/lib/adt_testbed.py b/lib/adt_testbed.py
index 7fc1158..a0b6a67 100644
--- a/lib/adt_testbed.py
+++ b/lib/adt_testbed.py
@@ -499,6 +499,7 @@ Description: satisfy autopkgtest test dependencies
         # so we might need to retry without pinning
         download_fail_retries = 3
         while True:
+            self.check_exec(['mount', '-o', 'remount,rw', '/'])
             self.check_exec(['dpkg', '--unpack', deb.tb], stdout=subprocess.PIPE)
             # capture status-fd to stderr
             (rc, _, serr) = self.execute(['/bin/sh', '-ec', '%s apt-get install '
-- 
2.10.1

